src/TRoom.cpp

```diff
@@ -691,12 +692,33 @@ void TRoom::restore(QDataStream& ifs, int roomID, int version)
     if (version >= 6) {
         ifs >> other;
     }
-    if (version >= 9) {
-        ifs >> c;
+
+    qint8 oldCharacterCode = 0;
+    if (version >= 19) {
```

Comment:
If we can use userdata for this information, why introduce a new map version that instantly makes it incompatible with all older mudlets?

This is less of a pain point now with automated updates, but still, why cause this pain to people for every minor map upgrade? :thinking:

Reply:
Because leaving the data in the user-data area is fragile - they could wipe it/modify accidentally and less code friendly internally as it has to be store as a string (admittedly that is less of an issue for __this__ feature)...

Reply:
There's a ton of ways people can mess up the map by accident and I think editing the userdata is the least of them. A more practical problem is selecting and moving the rooms accidentally and there being no "undo" operation for that.